src/api/views/templates.py

```diff
@@ -23,33 +24,58 @@ def get(self):
         return jsonify(template_manager.all())
 
     def post(self):
-        """Create new template."""
-        category = request.args.get("category")
-
-        resp = requests.post(
-            f"{STATIC_GEN_URL}/template/?category={category}",
-            files={"zip": (f"{category}.zip", request.files["zip"])},
-        )
-
-        try:
-            resp.raise_for_status()
-        except requests.exceptions.HTTPError:
-            return jsonify({"error": resp.text}), 400
+        """Create new template."""                
+        overwrite = request.args.get("overwrite")
+        bOverwrite = overwrite.lower().strip()=="true"
+        def checkName(name):                    
+            templates = template_manager.all()                    
+            existing_names = [(template["name"],template["_id"]) for template in templates if name == template["name"]]        
+            return len(existing_names) > 0, existing_names
+            
+        def cleanup(name, document_id):            
+            resp = requests.delete(f"{STATIC_GEN_URL}/template/?category={name}")            
+            try:
+                resp.raise_for_status()
+            except requests.exceptions.HTTPError as e:
+                return {"name":name, "status": str(e)}             
+            template_manager.delete(document_id)
+            return {"name":name, "status":"deleted"}
+        
+
+        rvalues = []               
+        for file in request.files.getlist('zip'):
+            if file.filename.endswith(".zip"):
+                name = file.filename[:-4]
+            (exists,ids) = checkName(name)
+
+            if(bOverwrite and exists):
+                cleanup(name,ids[0][1])
+                exists = False
+            
+            if(not exists):
+                urlEscapedName = urllib.parse.quote_plus(name)
+                resp = requests.post(
+                    f"{STATIC_GEN_URL}/template/?category={urlEscapedName}",
+                    files={"zip": (f"{file.filename}", file)},
+                )
+                try:
+                    resp.raise_for_status()
+                except requests.exceptions.HTTPError as e:
+                    return jsonify({"error": str(e)})
 
-        # remove temp files
-        shutil.rmtree("tmp/", ignore_errors=True)
+                # remove temp files
+                shutil.rmtree("tmp/", ignore_errors=True)
```

Comment:
Should we be removing everything in tmp? Or just the files in the path that were downloaded/uploaded?